packages/store/addon/-private/system/core-store.ts

```diff
@@ -3111,7 +3116,26 @@ abstract class CoreStore extends Service {
     clientId: string,
     storeWrapper: RecordDataStoreWrapper
   ): RecordData {
-    throw new Error(`Expected store.createRecordDataFor to be implemented but it wasn't`);
+    if (HAS_RECORD_DATA_PACKAGE) {
+      // we can't greedily use require as this causes
+      // a cycle we can't easily fix (or clearly pin point) at present.
```

Comment:
```suggestion
      // a cycle we can't easily fix (or clearly pin point) at present.
      //
      // it can be reproduced in partner tests by running
      // node ./bin/packages-for-commit.js && cd ../__external-test-cache/ember-observer && ember test
```